### ISSUE TITLE
Plots

### ISSUE THREAD

**Issue by Freshmilkymilk** [ArdaCraft/IssueTracker#18]
Add a plots world.

**Comment by Freshmilkymilk** [comment]
**Thoughts:**

Multiworld would probably be the best solution.

- Reduces server usage, as there won't be an entire server to take up resources.
- Updating things is easier than another server instance
- Easier to handle cross 'world' chat

Plugins we can use:

1. [Project Worlds](https://forums.spongepowered.org/t/project-worlds-v0-9-1-world-manager/)
2. [PlotSquared](https://forums.spongepowered.org/t/release-plotsquared-grief-protection-multi-world-management-world-generators-survival-creative/)

Things that would be good to have:

- Merged plots
- WorldEdit restricted to plots
- Make the spawn area of plots a sort of example area (Section off area)

**Comment by dags-** [comment]
Multiworld vs Server Instance is a tricky one imo.

### Performance:
You'd probably use up less ram with multiworld, however, you'd probably see better cpu load balancing by running separate server instances. Given that ram is 'cheap' relative to cpus, it's probably better to aim to keep cpu load down at the expense of ram, rather than visa versa.

### Simplicity:
The main benefit of multiworld is simplicity. Basically you don't need to worry about keeping blockpacks, mods, plugins in sync, nor rely on specialized plugins to handle chat/teleportation between worlds.

Also, there shouldn't be any issue with block id miss-matches when transferring schematics from world to world - I think it'd be possible to avoid such issues with separate server instances, but might require a little extra work

**Comment by Freshmilkymilk** [comment]
After testing a bit:

## Multiworld
To have a multiworld plot solution, `allow-nether` has to be set to true. Currently our setup doesn't have this. When changing to true the server changes all the block ids, making the world unusable.

## Server Instance
We'll have to use this as a solution.

There are a few options for "bungee-cord" within sponge. Sponge already supports bungee-cord, but there are a few plugins that simplify bungee's api.
[SpongyCord](https://ore-staging.spongepowered.org/pie_flavor/SpongyCord)
[SpongePls](https://forums.spongepowered.org/t/spongepls-bungeecord-forge-support/)

**Comment by dags-** [comment]
Some more links

A more actively updated/developed fork of bungeecord:
https://github.com/WaterfallMC/Waterfall

A new(er) proxy server written in Java or Go, probably performs better - not sure if it supports proxy plugins, or Sponge for that matter:
https://github.com/LilyPad/JLilyPad
https://github.com/LilyPad/GoLilyPad

**Comment by HenrySlawniak** [comment]
The `allow-nether` setting doesn't make sense, Sponge has APIs for generating and switching between worlds. Was there a specific plot plugin you were looking at?

**Comment by dags-** [comment]
He was using [PlotSquared](https://github.com/IntellectualSites/PlotSquared). I'm having a look at whether building it against a newer version of sponge helps, looks to be failing to run scheduled tasks with our version of spongeforge atm.

**Comment by dags-** [comment]
Ah, it is a Sponge requirement: https://github.com/SpongePowered/SpongeForge/issues/287

Anyways, I was able to set it to true without the block ids getting messed up.
I used the world from 'world.2016-08-08_00_00.tar.gz' and the rest of the setup was pulled straight from the dev server dir.
I ran the server once to make sure everything loaded normally before adding the plots plugin in, and creating a basic plots world.
